spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/deployer/ReleaseAnalyzer.java

```diff
@@ -101,45 +100,57 @@ private ReleaseAnalysisReport analyzeDependentPackagesOnly(
-			ReleaseDifference difference = compare(existingApplicationManifest,
-					matchingReplacingApplicationManifest);
-			if (!difference.areEqual()) {
-				logger.info(
-						"Dependent package difference found for " + existingApplicationManifest.getApplicationName());
-				appsToDelete.add(existingApplicationManifest.getApplicationName());
-				diffMessagesBuilder.append(difference.getDifferenceSummary());
-				diffMessagesBuilder.append("\n");
-			}
-		}
-		if (appsToDelete.size() != 0) {
-			ReleaseDifference releaseDifference = new ReleaseDifference(false, diffMessagesBuilder.toString());
-			return new ReleaseAnalysisReport(appsToDelete, releaseDifference, existingRelease, replacingRelease);
-		}
-		else {
-			ReleaseDifference releaseDifference = new ReleaseDifference(true);
-			return new ReleaseAnalysisReport(new ArrayList<>(), releaseDifference, existingRelease, replacingRelease);
+					applicationName, replacingApplicationSpecList);
+			ApplicationManifestDifference applicationManifestDifference = applicationManifestDifferenceFactory
+					.createApplicationManifestDifference(applicationName,
+							existingApplicationManifest,
+							matchingReplacingApplicationManifest);
+			applicationManifestDifferences.add(applicationManifestDifference);
 		}
+
+		return createReleaseAnalysisReport(existingRelease, replacingRelease, appsToDelete,
+				applicationManifestDifferences);
+
 	}
 
 	private ReleaseAnalysisReport analyzeTopLevelPackagesOnly(
 			List<? extends SpringCloudDeployerApplicationManifest> existingApplicationSpecList,
 			List<? extends SpringCloudDeployerApplicationManifest> replacingApplicationSpecList,
 			Release existingRelease, Release replacingRelease) {
-		ReleaseDifference difference = compare(existingApplicationSpecList.get(0),
-				replacingApplicationSpecList.get(0));
+
 		List<String> appsToDelete = new ArrayList<>();
-		if (!difference.areEqual()) {
-			logger.info(
-					"Differences detected, upgrading app " + existingApplicationSpecList.get(0).getApplicationName());
-			appsToDelete.add(existingApplicationSpecList.get(0).getApplicationName().trim());
-		}
-		return new ReleaseAnalysisReport(appsToDelete, difference, existingRelease, replacingRelease);
+		List<ApplicationManifestDifference> applicationManifestDifferences = new ArrayList<>();
+
+		ApplicationManifestDifference applicationManifestDifference = applicationManifestDifferenceFactory
+				.createApplicationManifestDifference(
+						existingApplicationSpecList.get(0).getApplicationName(),
+						existingApplicationSpecList.get(0),
+						replacingApplicationSpecList.get(0));
+		applicationManifestDifferences.add(applicationManifestDifference);
+
+		return createReleaseAnalysisReport(existingRelease, replacingRelease, appsToDelete,
+				applicationManifestDifferences);
+	}
+
+	private ReleaseAnalysisReport createReleaseAnalysisReport(Release existingRelease,
+			Release replacingRelease,
+			List<String> appsToDelete,
```

Comment:
Can better change `appsToDelete` -> something like  `appsToUpdate`

Reply:
Yes, also depends on the results of the previous comment if it should stick around at all.